.github/workflows/template-maintenance.yml

```diff
@@ -0,0 +1,262 @@
+name: Template Maintenance
+
+on:
+  workflow_run:
+    workflows: ["Release Please"]
+    types:
+      - completed
+    branches:
+      - main
+
+# Prevent multiple simultaneous runs
+concurrency:
+  group: template-maintenance-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+  pull-requests: write
+  actions: read
+
+jobs:
+  detect-release:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    outputs:
+      should-upgrade: ${{ steps.check-release.outputs.should-upgrade }}
+      release-tag: ${{ steps.check-release.outputs.release-tag }}
+      release-type: ${{ steps.check-release.outputs.release-type }}
+      release-url: ${{ steps.check-release.outputs.release-url }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.TAMBO_UPGRADE_PAT || secrets.GITHUB_TOKEN }}
+
+      - name: Make detection script executable
+        run: chmod +x .github/scripts/detect-release.sh
+
+      - name: Check for relevant release
+        id: check-release
+        env:
+          GITHUB_REPOSITORY: ${{ github.repository }}
+        run: .github/scripts/detect-release.sh
```

Comment:
I actually think there might be a way to capture the outputs of the Release Please workflow in this workflow, so you don't need a separate script to detect? I'm not 100% sure on that though